arviz/tests/base_tests/test_stats.py

```diff
@@ -47,6 +47,35 @@ def test_hpd():
     assert_array_almost_equal(interval, [-1.88, 1.88], 2)
 
 
+def test_hpd_2darray():
+    normal_sample = np.random.randn(12000, 5)
+    result = hpd(normal_sample)
+    assert result.shape == (5, 2,)
+
+
+def test_hpd_multidimension():
+    normal_sample = np.random.randn(12000, 10, 3)
+    result = hpd(normal_sample)
+    assert result.shape == (10, 3, 2,)
+
+
+def test_hpd_idata():
+    data = load_arviz_data("centered_eight")
+    normal_sample = data.posterior
+    result = hpd(normal_sample)
+    assert isinstance(result, Dataset)
+    assert result.dims == {"school": 8, "hpd": 2}
+
+    result = hpd(normal_sample, **{"input_core_dims": [["chain"]]})
+    assert isinstance(result, Dataset)
+    assert result.dims == {"draw": 500, "hpd": 2, "school": 8}
+
+    result = hpd(normal_sample, var_names=["mu", "theta"])
+    assert isinstance(result, Dataset)
+    assert result.dims == {"hpd": 2, "school": 8}
+    assert list(result.data_vars.keys()) == ["mu", "theta"]
+
```

Comment:
to test `group` argument you could use something like:

```
def test_hpd_idata_group(centered_eight):
    result_posterior = hpd(centered_eight, group="posterior", var_names="mu")
    result_prior = hpd(centered_eight, group="prior", var_names="mu")
    assert result_prior.dims == {"hpd": 2}
    assert result_posterior.mu.values[0] > result_prior.mu.values[0]
    assert result_posterior.mu.values[1] < result_prior.mu.values[1]
```

Not really sure about the values and indexes, please check. The posterior distribution should be more constrained than the prior, therefore the posterior hpd should lie inside the prior hpd (which is what this tries to check)

Reply:
Okay, I understand the logic. 
But, I am getting `mu.values` for posterior to be [-2.11840595 10.4032071 ]  and for prior to be [-9.93081274  8.76928874]. (not completely inside prior)
> assert result_posterior.mu.values[1] < result_prior.mu.values[1]

So, this is failing.?

Reply:
The region in the prior is much larger, but centered around 0 whereas the region in the posterior is smaller but not centered around zero, therefore, even though it is smaller than the prior hpd it does not lie inside prior hpd region, you can change it to some other check that actually works. I was also thinking that `sel` should also be added, otherwise users can select theta variable but nothing else.

Reply:
Okay I understand. Can I check the range of prior and posterior? (By doing `result_posterior.mu.values[1] - result_posterior.mu.values[0]`)

Are you suggesting to add new `sel` param?

Reply:
You can check the range yes. And yes, I am suggesting to add a sel argument